src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -235,180 +276,215 @@ protected override AuthenticationTicket AuthenticateCore()
             {
                 IFormCollection form = await Request.ReadFormAsync();
```

Comment:
Ah. While we have formalized the request buffering and caching behavior, we haven't officially decided who is responsible for re-winding the request stream after it's read and buffered. Even if ReadFormAsync caches the parsed form, that doesn't help another API like Nancy that may try to read the stream directly. I think this line should move into ReadFormAsync, but that hasn't happened yet.